### ISSUE TITLE
LTI 1.1 - OAuth Parameter `oauth_callback`  should be optional

### ISSUE THREAD

**Issue by dennis531** [jupyterhub/ltiauthenticator#136]
### Proposed change

An optional `oauth_callback` parameter would be preferable, since not every LTI tool consumer provides this parameter. Also according to https://oauth.net/core/1.0/#rfc.section.6.2.1 this parameter is optional. For example, [VIPS](https://develop.studip.de/studip/plugins.php/pluginmarket/presenting/details/81097da5ef66a002998b75d5eeece1f0) doesn't provide this parameter. The proposed solution in https://www.imsglobal.org/specs/ltiv1p1/implementation-guide#toc-2 to set an URL parameter to `oauth_callback=about:blank` would be a high failure factor in a Learning Management System like [Stud.IP](https://www.studip.de/) for lecturers. 

Furthermore, I don't see any reason why JupyterHub would require this parameter.

### Who would use this feature?

This could be anyone who has a LTI consumer that doesn't deliver this parameter.

**Comment by martinclaus** [comment]
Hi @dennis531! Indeed, the `oauth_callback` parameter seems to be obsolete in the login flow used by LTI 1.1 and it is not required anywhere.

It would be great if you could provide the anonymised parameters send by your tool consumer in the LTI launch message. You should see them in the logs when [log level is set to "DEBUG"](https://jupyterhub.readthedocs.io/en/stable/reference/config-reference.html). This would help to understand the edge-cases that needs to be handled.

**Comment by dennis531** [comment]
Here are my logs. I hope it helps:

```
python3[1301769]: [D 2023-03-06 08:31:14.245 JupyterHub auth:139] Decoded args from request: 
{
  'lti_version': 'LTI-1p0', 
  'lti_message_type': 'basic-lti-launch-request', 
  'oauth_consumer_key': 'consumer_key',   # modified
  'oauth_version': '1.0', 
  'oauth_nonce': 'lti6405a4d2298351.38731435', 
  'oauth_timestamp': '1678091474', 
  'oauth_signature_method': 'HMAC-SHA1', 
  'tool_consumer_info_product_family_code': 'studip', 
  'tool_consumer_info_version': '5.3.alpha-svn', 
  'tool_consumer_instance_guid': 'studip', 
  'tool_consumer_instance_name': 'Stud.IP', 
  'tool_consumer_instance_description': 'Stud.IP 5.1 - Studienbegleitender Internetsupport von Präsenzlehre', 
  'tool_consumer_instance_url': 'http://localhost/studip/', 
  'tool_consumer_instance_contact_email': 'studip@studip', 
  'resource_link_id': '10', 
  'resource_link_title': 'Jupyter Notebook', 
  'resource_link_description': 'Teste den Code!', 
  'context_id': '07db7bd8e8e291b58e2c233d77c404bc', 
  'context_type': 'CourseSection', 
  'context_label': '', 
  'context_title': 'JupyterHub-Test', 
  'user_id': '76ed43ef286fb55cf9e41beadb484a9f', 
  'roles': 'Learner', 
  'launch_presentation_locale': 'de-DE', 
  'launch_presentation_document_target': 'iframe', 
  'custom_next': '/hub/user-redirect/tree/Untitled.ipynb', 
  'lis_outcome_service_url': 'http://localhost/studip/plugins.php/vipsplugin/sheets/relay/outcome?exercise_id=10', 
  'lis_result_sourcedid': '2:10:76ed43ef286fb55cf9e41beadb484a9f', 
  'oauth_signature': '04Mdzu/0Yu7eK9xszcEbKxPAEN8='
}
```

**Comment by martinclaus** [comment]
@dennis531 Since I am lacking a test system and I got the impression that you have one set up already: would it be possible for you to test the above mentioned PR?

**Comment by martinclaus** [comment]
Fixed in 91b713079ed120eefac420f50db08f887b36065a.
